src/commands/client/namespace/index.ts

```diff
@@ -58,6 +59,32 @@ export class Namespace<TIndexMetadata extends Dict = Dict> {
     args: CommandArgs<typeof UpsertCommand<TMetadata>>
   ) => new UpsertCommand<TMetadata>(args, { namespace: this.namespace }).exec(this.client);
 
+  /*
+   * Updates specific items in the index.
+   * It's used for updating existing items in the index.
+   *
+   * @example
+   * ```js
+   * const updateArgs = {
+   *   id: '123',
+   *   vector: [0.42, 0.87, ...],
+   *   metadata: { property1: 'value1', property2: 'value2' }
+   * };
+   * const updateResult = await index.update(updateArgs);
+   * console.log(updateResult); // Outputs the result of the update operation
+   * ```
+   *
+   * @param {CommandArgs<typeof UpdateCommand>} args - The arguments for the update command.
+   * @param {number|string} args.id - The unique identifier for the item being updated.
+   * @param {number[]} args.vector - The feature vector associated with the item.
+   * @param {Record<string, unknown>} [args.metadata] - Optional metadata to be associated with the item.
+   *
+   * @returns {Promise<{updated: number}>} A promise that resolves with the result of the update operation after the command is executed.
```

Comment:
Instead of "result of the update operation", maybe something more descriptive like, "the number of updated vectors" would be better.